src/App.tsx

```diff
@@ -1,204 +1,90 @@
+/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-unused-vars */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { USER_ID, getTodos } from './api/todos';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { TodoAppHeader } from './components/TodoAppHeader';
+import { TodoAppMain } from './components/TodoAppMain';
+import { TodoAppFooter } from './components/TodoAppFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Todo } from './types/Todo';
+import { ErrorText } from './types/ErrorText';
+import { StatusFilter } from './types/StatusFilter';
+import { wait } from './utils/fetchClient';
+
+const getFilteredTodos = (
+  todos: Todo[],
+  statusFilter: StatusFilter,
+): Todo[] => {
+  switch (statusFilter) {
+    case StatusFilter.All:
+      return todos;
+    case StatusFilter.Active:
+      return todos.filter(todo => !todo.completed);
+    case StatusFilter.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorText, setErrorText] = useState<ErrorText>(ErrorText.NoError);
+  const [statusFilter, setStatusFilter] = useState<StatusFilter>(
+    StatusFilter.All,
+  );
+
+  useEffect(() => {
+    getTodos().then(
+      data => {
+        wait(150).then(() => setTodos(data));
+      },
+      () => {
+        setErrorText(ErrorText.Loading);
+        wait(3000).then(() => setErrorText(ErrorText.NoError));
```

Comment:
Create a handler that will do it for you as you have to do it every time when you call some API
```typescript
const handleError = (message: ErrorText) => {
  setErrorText(message);
  wait(3000).then(() => setErrorText(ErrorText.NoError));
};
```